src/runtime_src/core/common/module_loader.cpp

```diff
@@ -146,6 +147,29 @@ shim_path()
   return path;
 }
 
+static std::vector<std::string>
+driver_plugin_paths()
+{
+  std::vector<std::string> ret;
+  std::vector<bfs::path> vec{
+    bfs::directory_iterator(shim_path().parent_path()),
+    bfs::directory_iterator()
+  };
```

Comment:
Do we really need a heap allocated vector for this?   
Other patterns in XRT use a while-loop, I see the appeal of this a set that can be iterated, new pattern to me for sure, but you should use std::array in this case.

Reply:
I think a while-loop would do. Will remove this vector.